src/sys.d.ts

```diff
@@ -0,0 +1,1352 @@
+  basename(): string
+  ext(): string | null
+  mimeType(): MimeType | null
+  exists(): boolean
+  size(): number | null
+  isEmpty(): boolean
+  isHidden(): boolean
+  isReadable(): boolean
+  isWritable(): boolean
+  isExecutable(): boolean
+  osPath(): string | null
+  parent(): File | null
+  list(pattern?: Regex | null): List<File>
+  listDirs(pattern?: Regex | null): List<File>
+  listFiles(pattern?: Regex | null): List<File>
+  walk(c: ((argA: File) => void)): void
+  normalize(): File
+  plus(path: Uri, checkSlash?: boolean): File
+  store(): FileStore
+  create(): File
+  createFile(name$: string): File
+  createDir(name$: string): File
+  copyTo(to: File, options?: Map<string, JsObj> | null): File
+  copyInto(dir: File, options?: Map<string, JsObj> | null): File
+  moveTo(to: File): File
+  moveInto(dir: File): File
+  rename(newName: string): File
+  delete$(): void
+  deleteOnExit(): File
+  open(mode?: string): Buf
+  mmap(mode?: string, pos?: number, size?: number | null): Buf
+  in$(bufferSize?: number | null): InStream
+  out(append?: boolean, bufferSize?: number | null): OutStream
+  readAllBuf(): Buf
+  readAllLines(): List<string>
+  eachLine(f: ((argA: string) => void)): void
+  readAllStr(normalizeNewlines?: boolean): string
+  readProps(): Map<string, string>
+  writeProps(props: Map<string, string>): void
+  readObj(options?: Map<string, JsObj> | null): JsObj | null
+  writeObj(obj: JsObj | null, options?: Map<string, JsObj> | null): void
+}
+export class InterruptedErr extends Err {
+}
+export class Env extends Obj {
+  static cur(): Env
+  parent(): Env | null
+  platform(): string
+  os(): string
+  arch(): string
+  runtime(): string
+  javaVersion(): number
+  idHash(obj: JsObj | null): number
+  args(): List<string>
+  mainMethod(): Method | null
+  vars(): Map<string, string>
+  diagnostics(): Map<string, JsObj>
+  gc(): void
+  host(): string
+  user(): string
+  in$(): InStream
+  out(): OutStream
+  err(): OutStream
+  prompt(msg?: string): string | null
+  promptPassword(msg?: string): string | null
+  homeDir(): File
+  workDir(): File
+  tempDir(): File
+  path(): List<File>
+  findFile(uri: Uri, checked?: boolean): File | null
+  findAllFiles(uri: Uri): List<File>
+  findPodFile(podName: string): File | null
+  findAllPodNames(): List<string>
+  compileScript(f: File, options?: Map<string, JsObj> | null): Type
+  compileScriptToJs(f: File, options?: Map<string, JsObj> | null): string
+  index(key: string): List<string>
+  indexKeys(): List<string>
+  indexPodNames(key: string): List<string>
+  props(pod: Pod, uri: Uri, maxAge: Duration): Map<string, string>
+  config(pod: Pod, key: string, def?: string | null): string | null
+  locale(pod: Pod, key: string, def?: string | null, locale?: Locale): string | null
+  exit(status?: number): void
+  addShutdownHook(hook: (() => void)): void
+  removeShutdownHook(hook: (() => void)): boolean
+}
+export class Map<K = unknown, V = unknown> extends Obj {
```

Comment:
For collection types, would be nice to implement iterators: https://dev.to/gsarciotto/iterators-in-typescript-1d78
This would require JS code generation also.